cluster-api/cloud/google/machineactuator.go

```diff
@@ -59,12 +60,12 @@ type SshCreds struct {
 }
 
 type GCEClient struct {
-	service       *compute.Service
-	scheme        *runtime.Scheme
-	codecFactory  *serializer.CodecFactory
-	kubeadmToken  string
-	sshCreds      SshCreds
-	machineClient client.MachineInterface
+	computeService clients.ComputeService
```

Comment:
The interface should be defined here. This is the consumer of the functionality and is best aware of what functionality it needs.